pkg/storage/fs/ocis/blobstore/blobstore.go

```diff
@@ -48,20 +48,42 @@ func New(root string) (*Blobstore, error) {
 }
 
 // Upload stores some data in the blobstore under the given key
-func (bs *Blobstore) Upload(node *node.Node, data io.Reader) error {
+func (bs *Blobstore) Upload(node *node.Node, source string) error {
 
 	// ensure parent path exists
 	if err := os.MkdirAll(filepath.Dir(bs.path(node)), 0700); err != nil {
 		return errors.Wrap(err, "Decomposedfs: oCIS blobstore: error creating parent folders for blob")
 	}
 
-	f, err := os.OpenFile(bs.path(node), os.O_CREATE|os.O_WRONLY, 0700)
+	// check that destination is not a directory, that makes Rename fail
```

Comment:
Otherwise I think I would rather omit the special handling for the non-expected case, knowing that the code writing the blob will fail anyway when trying to write to a directory.

Reply:
Ok, removed the directory check.